packages/backend/src/clients/Aws/s3.ts

```diff
@@ -50,6 +50,22 @@ export class S3Service {
         return data.Location;
     }
 
+    async uploadCsv(csv: string, csvName: string): Promise<string> {
```

Comment:
Instead of making the file public, we can upload it and return a link that is only valid for an hour or so with a presigned URL. That way we don't need to be concerned about cleaning up long-lived public .csv files:

```
// First put object and make private

// then generate url
const url = S3.getSignedUrl('getObject', {
    Bucket: this.lightdashConfig.s3.bucket,
    Key: csvName,
    Expires: 3600, // an hour
})
```

Reply:
Good idea, let me try that

Reply:
Done